projects/igniteui-angular/src/lib/directives/autocomplete/autocomplete.directive.ts

```diff
@@ -0,0 +1,305 @@
+        return this.elementRef.nativeElement;
+    }
+
+    /**
+     * @hidden
+     */
+    get parentElement(): HTMLElement {
+        return this.group ? this.group.element.nativeElement : this.nativeElement;
+    }
+
+    /**
+     * @hidden
+     */
+    @Input('igxAutocomplete')
+    public dropDown: IgxDropDownComponent;
+
+    /**
+     * Enables/disables autocomplete component
+     *
+     * ```typescript
+     * // get
+     * let disabled = this.autocomplete.disabled;
+     * ```
+     * ```html
+     * <!--set-->
+     * <input type="text" [igxAutocomplete]="townsPanel" [igxAutocompleteDisabled]="disabled"/>
+     * ```
+     * ```typescript
+     * // set
+     * public disabled = true;
+     * ```
+     */
+    @Input('igxAutocompleteDisabled')
+    get disabled() {
+        return this._disabled;
+    }
+    set disabled(value) {
+        this._disabled = value;
+        this.close();
+    }
+
+    /**
+     * Provide overlay settings for the autocomplete drop down
+     *
+     * ```typescript
+     * // get
+     * let settings = this.autocomplete.autocompleteSettings;
+     * ```
+     * ```html
+     * <!--set-->
+     * <input type="text" [igxAutocomplete]="townsPanel" [igxAutocompleteSettings]="settings"/>
+     * ```
+     * ```typescript
+     * // set
+     * this.settings = {
+     *  positionStrategy: new ConnectedPositioningStrategy({
+     *      closeAnimation: null,
+     *      openAnimation: null
+     *  })
+     * };
+     * ```
+     */
+    @Input('igxAutocompleteSettings')
+    autocompleteSettings: AutocompleteOverlaySettings;
+
+    /**
+     * Emitted after item from the drop down is selected
+     *
+     * ```html
+     * <input igxInput [igxAutocomplete]="townsPanel" (onItemSelected)='itemSelected()' />
+     * ```
+     */
+    @Output()
+    onItemSelected = new EventEmitter<any>();
+
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.autocomplete')
+    public autofill = 'off';
+
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.role')
+    public role = 'combobox';
+
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.aria-expanded')
+    public get ariaExpanded() {
+        return !this.collapsed;
+    }
+
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.aria-haspopup')
+    public get hasPopUp() {
+        return 'listbox';
+    }
+
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.aria-owns')
+    public get ariaOwns() {
+        return this.dropDown.id;
+    }
+
+    /**
+     * @hidden
+     */
+    @HostListener('input', ['$event'])
+    onInput() {
+        if (this.disabled)  {
+            return;
+        }
+        if (this.collapsed) {
+            this.open();
+        } else {
+            this.unhighlightFirstItem();
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    @HostListener('blur', ['$event'])
+    onBlur() {
+        this.close();
+    }
+
+    /**
+     * @hidden
+     */
+    handleFocus() {}
```

Comment:
Is this some override?

Reply:
It was but now it is not needed. Thanks!